stack_data/core.py

```diff
@@ -387,6 +410,28 @@ def render(
         return ''.join(parts)
 
 
+class EmptyLine(Line):
```

Comment:
Does this class need to exist? Is it just an optimisation of `render`?

Reply:
When I tried to simply use `Line`, an exception was raised. This was the minimal subclass that worked.

Reply:
That's surprising, why didn't Line work?

Reply:
Good question ... I just tried it, and it worked this time.

I'll make the other suggested changes quickly and will commit a new version.  

I'll investigate this, and start looking at adding unit tests after I finish work today.

Reply:
I found out why using `Line` was raising an exception before; let's just say it was due to a user error.
I have not had time to add unit tests, but everything else should be done.